### PR TITLE
[Research Foundation] Protolathe & Circuit Imprinter science functionality

### PR DESCRIPTION
**Pull request by AtnerNT**
Addresses #8164 

Adds new functionality to the protolathe and circuit imprinter, the item menus of the machines are now dependent on the technologies researched on its connected research server as well as their assigned initial designs. They can also be assigned a department to restrict the designs they have access too.
A R&DProduction Machine (such as a protolathe or CI) can be connected to a research server via a multitool.
A R&DProduction machine can function without a connected server but will only have access to it's initial design list.

NOTES:
Due to the techwebs W.I.P state a few temporary measures are in effect:
-Right now the circuit imprinter starts with all the designs that the old CI could produce (around 62 designs). This is so gameplay is not effected while players cannot research tech.
-Once the techweb harddrive is merged some minor changes will have to be made so this still works and I have left some comments in ResearchServer.cs to indicate some things that should be added once the techweb is complete
-There are a good hundred or two designs currently in the game, and due to the monotonous nature of the task not all of them have been updated with an accurate Item_ID for this PR, only the ones needed by the CI and Protolathe (+a few others) have been done. I will correct all the designs in a future pr once I have an hour or two to waste copy pasting values.
-Lastly, as it was not part of the issue- the ExoSuit Fabricator does not work off of the new system yet. However when its time it shouldn't be too hard of a transition and Ill probably work on moving it over once this PR has been properly tested on a live build if no one beats me to it first.

CL: [New] (Research Foundation) Protolathe & Circuit Imprinter science functionality. The recipes they have can now be updated with new recipes researched by scientists.

### CODE REVIEW

**Comment by AtnerNT** [comment]
Just saw some files that should not be in the changes.

File: UnityProject/Assets/Scripts/Systems/Research/Designs.cs [hunk 1]
```diff
@@ -60,11 +61,11 @@ public class Designs : MonoBehaviour
...
 	void Awake()
 	{
-		//if (!(Globals.IsInitialised))
-		//{
-		//	JsonImportInitialization();
-		//	Globals.IsInitialised = true;
-		//}
+		if (!(Globals.IsInitialised))
+		{
+			JsonImportInitialization();
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Systems/Research/Designs.cs]: this could be put in a task?

**Review by Bod9001 — DISMISSED** [review]
looks good

**Review by Bod9001 — APPROVED** [review]
looks good to merge I'll wait a little bit to see if anyone else has any changes wanted

**Pull request merged**